dotCMS/src/main/java/com/dotmarketing/business/VersionableFactoryImpl.java

```diff
@@ -278,92 +304,121 @@ protected void saveVersionInfo(VersionInfo info, boolean updateVersionTS) throws
-    	ContentletVersionInfo contv = null;
-    	 HibernateUtil dh = new HibernateUtil(ContentletVersionInfo.class);
-         dh.setQuery("from "+ContentletVersionInfo.class.getName()+" where identifier=? and lang=?");
-         dh.setParam(identifier);
-         dh.setParam(lang);
-         Logger.debug(this.getClass(), "getContentletVersionInfo query: "+dh.getQuery());
-         contv = (ContentletVersionInfo)dh.load();
-         return contv;
+    protected Optional<ContentletVersionInfo> findContentletVersionInfoInDB(String identifier, long lang)throws DotDataException, DotStateException {
+		final DotConnect dotConnect = new DotConnect()
+				.setSQL("SELECT * FROM contentlet_version_info WHERE identifier=? AND lang=?")
+				.addParam(identifier)
+				.addParam(lang);
+
+		List<ContentletVersionInfo> versionInfos = TransformerLocator
+				.createContentletVersionInfoTransformer(dotConnect.loadObjectResults()).asList();
+
+		return !versionInfos.isEmpty() ? Optional.of(versionInfos.get(0)) : Optional.empty();
     }
     @Override
     protected List<ContentletVersionInfo> findAllContentletVersionInfos(final String identifier)throws DotDataException, DotStateException {
+        final DotConnect dotConnect = new DotConnect()
+                .setSQL("SELECT * FROM contentlet_version_info WHERE identifier=?")
+                .addParam(identifier);
 
-         HibernateUtil dh = new HibernateUtil(ContentletVersionInfo.class);
-         dh.setQuery("from "+ContentletVersionInfo.class.getName()+" where identifier=? ");
-         dh.setParam(identifier);
+        List<ContentletVersionInfo> versionInfos = TransformerLocator
+                .createContentletVersionInfoTransformer(dotConnect.loadObjectResults()).asList();
 
-         Logger.debug(this.getClass(), "getContentletVersionInfo query: "+dh.getQuery());
-         return (List<ContentletVersionInfo>)dh.list();
+		return versionInfos==null || versionInfos.isEmpty()
+				? Collections.emptyList()
+				: versionInfos;
 
     }
     @Override
     protected void saveContentletVersionInfo(ContentletVersionInfo cvInfo, boolean updateVersionTS) throws DotDataException, DotStateException {
-    	Identifier ident = this.iapi.find(cvInfo.getIdentifier());
-    	ContentletVersionInfo vi= null;
-    	if(ident!=null && InodeUtils.isSet(ident.getId())){
-    		vi= findContentletVersionInfoInDB(ident.getId(), cvInfo.getLang());
-    	}
-    	boolean isNew = vi==null || !InodeUtils.isSet(vi.getIdentifier());
-        try {
-			BeanUtils.copyProperties(vi, cvInfo);
-		} catch (Exception e) {
-			throw new DotDataException(e.getMessage(),e);
+		boolean isNew = true;
+		if (UtilMethods.isSet(cvInfo.getIdentifier())) {
+			try {
+				final Optional<ContentletVersionInfo> fromDB =
+						findContentletVersionInfoInDB(cvInfo.getIdentifier(), cvInfo.getLang());
+				if (fromDB.isPresent()) {
+					isNew = false;
+				}
+			} catch (final Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6214071)